### PR TITLE
Fixed: server freeze and other SSL, error handling, and buffer bugs

### PR DESCRIPTION
**Pull request by pboothe**
The bug where the server would freeze sometimes. (accept() was getting interrupted)
  The bug where SSL sockets would cause the META test to fail.
  The bug where the OS name got cut off at 32 chars, almost certainly clobbering a pointer which was stored later in the same struct.
  readn_any's error handling, because SSL_read can return an error which literally means "please call SSL_read".
  run_meta_test_srv's error handling now correctly logs the error

### CODE REVIEW

File: src/test_meta_srv.c [hunk 1]
```diff
@@ -90,7 +90,7 @@ int test_meta_srv(Connection *ctl, tcp_stat_agent *agent,
                          testOptions->connection_flags);
       if (err) {
         // message reading error
-        log_println(0, "Protocol error!");
+        log_println(0, "Protocol error! (%d)", err);
```
Comment (stephen-soltesz) [inline src/test_meta_srv.c]: Nit: err is int64_t (mismatched with format), and it looks like it only needs to be an int. Suggest changing the variable type.

File: src/logging.h [hunk 2]
```diff
@@ -21,7 +21,7 @@
 #define PROTOLOGFILE "web100srvprotocol.log"  // name of protocol log file
 #define PROTOLOGPREFIX "web100srvprotocol_"  // prefix for protocol log file
 #define PROTOLOGSUFFIX ".log"  // suffix for protocol validation log file
-#define FILENAME_SIZE 256
+#define FILENAME_SIZE 1024
```
Comment (stephen-soltesz) [inline src/logging.h]: Future TODO.

While trying to decide if this value was sufficient or not, I looked under a rock in writeMeta(). Please include the FILENAME_SIZE and the various constants used in writeMeta as part of "TODO: ensure every write to a char array will always result in a null-terminated string of less than the buffer size."

There are several assumptions in writeMeta that don't add up. I think this value (based on how its used) wants to be PATH_MAX from <linux/limits.h>, as do some of the static buffer sizes in writeMeta(). There is a convention of using the client DNS name in the file name of logs when reverse lookup is successful. But there is a mismatch between the maximum DNS name and NAME_MAX from <linux/limits.h> and the other filename elements added by ndt.

**Comment by stephen-soltesz** [comment]
LGTM.

Added two comments, and updated the PR subject.

File: src/test_meta_srv.c (hunk 1 above)
Comment (pboothe) [inline src/test_meta_srv.c]: Fixed.

**Pull request merged**